app/models/staged/petition_creator.rb

```diff
@@ -0,0 +1,62 @@
+module Staged
+  module PetitionCreator
+    def self.manager(params, request, stage, move)
+      Manager.new(params, request, stage, move, self::Stages)
+    end
+
+    def self.stages
+      self::Stages.stage_names
+    end
+
+    class Manager
+      def initialize(params, request, stage, move, stages)
+        @params = params
+        @request = request
+        @previous_stage = stage
+        @move = move
+        @stages = stages
+      end
+
+      # This is the stage we came from - the UI elements we showed the user
+      # that generated these params
+      attr_reader :previous_stage
+      attr_reader :move
+      attr_reader :stages
+
+      def petition
+        @_petition ||= build_petition
+      end
+
+      def stage
+        stage_manager.result_stage.name
+      end
+
+      def stage_object
+        stage_manager.result_stage.stage_object
+      end
+
+      def create_petition
+        sanitize!
+        stage_manager.create
+      end
+
+      private
+
+      def stage_manager
+        @_stage_manager ||= Staged::StageManager.new(stages, previous_stage, move, petition)
+      end
+
+      def sanitize!
+        if petition.creator_signature
+          petition.creator_signature.email.strip! unless petition.creator_signature.email.blank?
```

Comment:
Had a think about this and it would work because:

1. calling ``to_s`` on a string returns that string, not a copy (this might be the kind of thing that changes between ruby stdlib versions)
2. calling ``to_s`` on ``nil`` to "cast" the value to a string to safely call ``strip!`` on it.

It's a neat trick, but I don't think it's as intention revealing as being explicit about what we're doing.  There might be a debate about performance though; perhaps always calling ``.to_s.strip!`` is faster (no conditional check) or maybe it's slower (always doing it even when we don't need to).  Either way it'd be a micro-optimisation.